### PR TITLE
Add servlet version check & javax, jakarta check in WebApp Handle Types Method

### LINKED ISSUE #12574: Update WebApp#addClassToHandlesTypesStartupSet to handle Jakarta Classes
Currently, it checks for java or javax and quickly returns.  Jakarta classes pass through, and this creates problems in some scenarios (websocket-2.0). These java, javax, and jakarta classes should not be initialized/instantiated. 

However, this may need to be revisited depending on what classes are within the customer's app.

### PR DESCRIPTION
**Pull request by volosied**
fixes #12574

It will be the customer's responsibility to transform their application before deploying it on a Jakarta enabled server. 

This is approach for this first release with the new namespace.  Allow customers to use the Transformer on their applications, or use their IDE to modify the source.  The dynamic aspect of modifying applications has been left as a future feature idea.

This PR may need to be revisited depending on what classes are within the customer's app.

### CODE REVIEW

**Comment by volosied** [comment]
#build

**Comment by volosied** [comment]
Build is good except for the following unrelated errors: 
`Caused by: java.lang.IllegalStateException: Previous attempts to find a Docker environment failed. Will not retry. Please see logs and check configuration`

**Comment by volosied** [comment]
#build

File: dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/webapp/WebApp.java [hunk 1]
```diff
@@ -1052,18 +1052,29 @@ protected void addClassToHandlesTypesStartupSet(String targetClassName, Set<Clas
...
       //might need to exclude something else like javax.* or java*
-      if (targetClassName.startsWith("java")) {
-          if ((targetClassName.charAt(4) == '.') || (targetClassName.startsWith("javax."))) {
-              if ( enableTrace ) { 
-                  logger.logp(Level.FINE, CLASS_NAME, methodName,
-                              "Internal class, {0}, is not added to the ServletContainerInitializers in the application: {1}",
-                              new Object[] { targetClassName, this.config.getDisplayName() });
-              }
-              return;
+      if(servletSpecLevel <= 40){
```
Comment (pnicolucci) [inline dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/webapp/WebApp.java:1058]: Can you use the constant instead: com.ibm.ws.webcontainer.osgi.WebContainer.SPEC_LEVEL_40

Comment (volosied) [reply dev/com.ibm.ws.webcontainer/src/com/ibm/ws/webcontainer/osgi/webapp/WebApp.java]: Changed

**Comment by volosied** [comment]
#run-libby-bot

**Pull request merged**